### PR TITLE
MINOR: Implement toString() in config validator classes

### PR DESCRIPTION
**Pull request by omkreddy**
### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by omkreddy** [comment]
@ijuma @rajinisivaram  minor change for doc rendering

**Comment by omkreddy** [comment]
also updated the log.dirs config description. Feel free to update the description as appropriate.

**Comment by rajinisivaram** [comment]
@omkreddy Should we add `toString()` to `CompositeValidator` and `NonEmptyStringWithoutControlChars` as well?

**Comment by omkreddy** [comment]
@rajinisivaram  Thanks for the review. currently above validators are not shown/used in the connect docs.  Added toString() method to NonEmptyStringWithoutControlChars. Not sure if it is required for  CompositeValidator.

**Comment by omkreddy** [comment]
retest this please

**Comment by omkreddy** [comment]
@ijuma @rajinisivaram can we merge this minor fix to 2.0?

File: core/src/main/scala/kafka/server/KafkaConfig.scala [hunk 1]
```diff
@@ -540,8 +540,8 @@ object KafkaConfig {
   val RackDoc = "Rack of the broker. This will be used in rack aware replication assignment for fault tolerance. Examples: `RACK1`, `us-east-1d`"
   /** ********* Log Configuration ***********/
   val NumPartitionsDoc = "The default number of log partitions per topic"
-  val LogDirDoc = "The directory in which the log data is kept (supplemental for " + LogDirsProp + " property)"
-  val LogDirsDoc = "The directories in which the log data is kept. If not set, the value in " + LogDirProp + " is used"
+  val LogDirDoc = "The directory in which the log data is kept (supplemental for " + LogDirsProp + " property). Default directory (" + Defaults.LogDir + ") is not safe for production deployment. Use valid persistent directory."
+  val LogDirsDoc = "The directories in which the log data is kept. If not set, the value in " + LogDirProp + " is used. Default directory (" + Defaults.LogDir + ") is not safe for production deployment. Use valid persistent directory."
```
Comment (ijuma) [inline core/src/main/scala/kafka/server/KafkaConfig.scala]: Is there a default? I thought the issue was the server.properties file.

Comment (omkreddy) [reply core/src/main/scala/kafka/server/KafkaConfig.scala]: We have default for "log.dir". There is no default for "log.dirs".  If "log.dirs" is not set, we will use "log.dir"

Comment (ijuma) [reply core/src/main/scala/kafka/server/KafkaConfig.scala]: You're right. It's even worse than I thought. It would be good to remove this unsafe default.

Comment (omkreddy) [reply core/src/main/scala/kafka/server/KafkaConfig.scala]: maybe we can merge this PR.  I will raise a minor KIP (if required) to remove the default .

**Comment by omkreddy** [comment]
@rajinisivaram @ijuma   Can we merge this minor PR. I reverted the change related to  "log.dirs" doc description.  I will handle "log.dirs" doc/default value related changes in separate PR.

**Comment by harshach** [comment]
+1.

**Review by rajinisivaram — APPROVED** [review]
@omkreddy Thanks for the PR, LGTM

**Comment by rajinisivaram** [comment]
Merged to trunk.

**Pull request merged**